tests/compile-test.rs

```diff
@@ -155,38 +155,37 @@ fn default_config() -> compiletest::Config {
     config
 }
 
-fn run_ui(cfg: &mut compiletest::Config) {
+fn run_ui(mut cfg: compiletest::Config) {
     cfg.mode = TestMode::Ui;
     cfg.src_base = Path::new("tests").join("ui");
+    cfg.build_base.push("ui");
```

Comment:
Hmm I don't like how `build_base` is modified in two places and folder name is repeated. Could we make it look like...
```rust
fn run_ui() {
    let mut config = base_config("ui");
```
so the `Config` is built from scratch. You can use a `SyncLazy` for `extern_flags()`. The rest of `default_config()` looks cheap enough to re-run with each test suite.

Reply:
Yeah that's much nicer, changed